pkg/azure/create.go

```diff
@@ -266,6 +267,26 @@ func createVirtualMachine(
 		return nil, err
 	}
 
+	// CustomData is cloud-init file or base64 encoded string
+	if azureProvider.Config.CustomData != "" {
+
+		// CustomData is base64 encoded string
+		if _, err := base64.StdEncoding.DecodeString(azureProvider.Config.CustomData); err == nil {
```

Comment:
Can we rephrase this if/else chain?

I think

if file -> file logic
else if NOT base64 -> error

Else we don't have anything to do